components/p3a/brave_p3a_service.cc

```diff
@@ -168,7 +166,7 @@ void BraveP3AService::Init() {
 
   // Init other components.
   uploader_.reset(new BraveP3AUploader(
-      g_browser_process->shared_url_loader_factory(), upload_server_url_,
+      shared_url_loader_factory, upload_server_url_,
```

Comment:
Curious why this changed is needed?
I think just passing `shared_url_url_loader_factory()` seems sufficient instead of having a reference to it in this `Init()` method.

Reply:
because using `g_browser_process` from components is a layering violation - https://github.com/brave/brave-core/pull/4277/files/31c731947bd357be66fe38c4cc07880ffed38e27#diff-f392158ed68d17529018865c90c7f8a0L30

Reply:
the other problem is timing. You can't access the shared_url_loader_factory when this class is initialized so it has to be passed in later